be/src/formats/orc/orc_mapping.cpp

```diff
@@ -0,0 +1,149 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Inc.
+
+#include "formats/orc/orc_mapping.h"
+
+namespace starrocks::vectorized {
+
+std::unique_ptr<OrcMapping> OrcMapping::build_mapping(const std::vector<SlotDescriptor*>& slot_descs,
+                                                      const orc::Type& root_orc_type, const bool case_sensitve) {
+    std::unique_ptr<OrcMapping> orc_mapping = std::make_unique<OrcMapping>();
+    Status res = _init_orc_mapping(orc_mapping, slot_descs, root_orc_type, case_sensitve);
+    if (!res.ok()) {
+        LOG(WARNING) << res.to_string();
+        return nullptr;
+    }
+    return orc_mapping;
+}
+
+size_t OrcMapping::get_column_id(size_t src_pos) {
+    auto it = _mapping.find(src_pos);
+    if (it == _mapping.end()) {
+        DCHECK(false);
+        // return large positive invalid number
+        return -1;
+    }
+    return it->second;
+}
+
+const OrcMappingPtr OrcMapping::get_child_mapping(size_t src_pos) {
+    auto it = _children.find(src_pos);
+    if (it == _children.end()) {
+        return nullptr;
+    }
+    return it->second;
+}
+
+void OrcMapping::add_mapping(size_t pos_in_src, size_t orc_column_id) {
+    _mapping.emplace(pos_in_src, orc_column_id);
+}
+
+void OrcMapping::add_child(size_t pos_in_src, const OrcMappingPtr& child_mapping) {
+    _children.emplace(pos_in_src, child_mapping);
+}
+
+void OrcMapping::clear() {
+    _mapping.clear();
+    _children.clear();
+}
+
+Status OrcMapping::_init_orc_mapping(std::unique_ptr<OrcMapping>& mapping,
+                                     const std::vector<SlotDescriptor*>& slot_descs, const orc::Type& orc_root_type,
+                                     const bool case_sensitve) {
+    // build tmp mapping for orc [orc field name -> pos in orc]
+    std::unordered_map<std::string, size_t> tmp_orc_fieldname_2_pos;
```

Comment:
it doesn't have to be prefixed a variable with `tmp_` instead of it's duplicate with another variable